python/love/producer/love_producer_csc.py

```diff
@@ -0,0 +1,552 @@
+from .producer_utils import get_data_type
+
+
+class LoveProducerCSC(LoveProducerBase):
+    """Specialized LOVE producer to deal with generic CSC behavior."""
+
+    def __init__(
+        self, domain: Domain, csc: str, log: Optional[logging.Logger] = None, **kwargs
+    ) -> None:
+        super().__init__(component_name=csc, log=log)
+
+        self.add_metadata(**kwargs)
+
+        self.remote: Remote = Remote(
+            domain,
+            csc,
+            index=kwargs.get("salindex", 0),
+            readonly=kwargs.get("remote_readyonly", True),
+        )
+
+        self._events_special_cases = {"evt_heartbeat"}
+
+        self._non_topic_data_stream = {}
+
+        self._revcode_topic_attribute_name_map: dict = dict()
+        self._template_manager_message: dict = dict()
+
+        self._need_reply_category = {"initial_state"}
+
+        self.periodic_data: list = self.get_periodic_data(**kwargs)
+        self.asynchronous_data: list = self.get_asynchronous_data(**kwargs)
+
+        self.store_last_sample_timeout: float = 0.5
+        self.heartbeat_timeout = 2.0
+        self.heartbeat_max_lost = 5
+
+        self.set_topic_name_revcode_mapping()
+        self.set_topic_template_manager_message_format()
+
+        self._heartbeat_monitor_task = None
+        self.done_task: asyncio.Future = asyncio.Future()
+
+        self.start_task: Awaitable = asyncio.create_task(self.start())
+
+    def should_reply_to_message_data(self, message_data: dict) -> bool:
+        """Determines whether a reply to message data should be sent.
+
+        Override base class default behavior.
+
+        Parameters
+        ----------
+        message_data: `dict`
+            Input dictionary with information used by the producer to determine
+            some action to take.
+
+        Returns
+        -------
+        `bool`
+            Should reply to message data?
+        """
+        category = message_data.get("category", None)
+        data = message_data.get("data", [dict()])[0]
+        csc = data.get("csc", None)
+        salindex = data.get("salindex", 0)
+
+        self.log.debug(
+            "Should reply to message_data: "
+            f"{category in self._need_reply_category}, "
+            f"{csc in self.reply_names}, "
+            f"{salindex == self.remote.salinfo.index}, "
+            f"{self.is_data_stream_stored(dict(stream=self.get_sample_name(message_data)))}. "
+            f"stream={self.get_sample_name(message_data)}"
+        )
+
+        return (
+            (category in self._need_reply_category)
+            and csc in self.reply_names
+            and salindex == self.remote.salinfo.index
+            and self.is_data_stream_stored(
+                dict(stream=self.get_sample_name(message_data))
+            )
+        )
+
+    def get_sample_name(self, data_stream: dict) -> str:
+        """Override base class default behavior."""
+        topic_name = data_stream["data"][0]["data"]["event_name"]
```

Comment:
I see that you considered the case described at: https://github.com/lsst-ts/LOVE-producer/pull/125/files#diff-bbacadd58a701e0ac0c02291b5a4a1512840be41e1b1ccc356d5335d13437648R184. So if I understand well, the `love_producer_base` has the basic behavior with the `stream` key and the `love_producer_csc` overrides the method using the `["data"]["event_name"]` accessor?

Reply:
exactly!